<img style="float: left; margin: 30px 15px 15px 15px;" src="https://pngimage.net/wp-content/uploads/2018/06/logo-iteso-png-5.png" width="300" height="500" /> 
    
    
### <font color='navy'> Simulación de procesos financieros. 

**Nombres:** Lyha Espinosa García, Carlos Daniel Pérez Hernández.

**Fecha:** 22 de Febrero del 2021.

**Expediente** : 721861, .
**Profesor:** Oscar David Jaramillo Zuluaga.

# Tarea 4: Clase 7

## Enunciado de tarea 
### Ejercicio 1
Como ejemplo simple de una simulación de Monte Carlo, considere calcular la probabilidad de una suma particular del lanzamiento de tres dados (cada dado tiene valores del uno al seis). Además cada dado tiene las siguientes carácterísticas: el primer dado no está cargado (distribución uniforme todos son equiprobables); el segundo y tercer dado están cargados basados en una distribución binomial con parámetros (`n=5, p=0.5` y `n=5, p=0.2`). Calcule la probabilidad de que la suma resultante sea 7, 14 o 18. 
### 2 Ejercicio de aplicación- Cafetería Central
Premisas para la simulación:
- Negocio de alimentos que vende bebidas y alimentos.
- Negocio dentro del ITESO.
- Negocio en cafetería central.
- Tipo de clientes (hombres y mujeres).
- Rentabilidad del 60%.

#### Objetivo
Realizar una simulación estimado el tiempo medio que se tardaran los clientes en ser atendidos entre el horario de 6:30 a 1 pm. Además saber el consumo. 
**Analizar supuestos y limitantes**
#### Supuestos en simulación
Clasificación de clientes: 
- Mujer  = 1 $\longrightarrow$ aleatorio < 0.5
- Hombre = 0 $\longrightarrow$ aleatorio $\geq$ 0.5.

Condiciones iniciales:
- Todas las distrubuciones de probabilidad se supondrán uniformes.
- Tiempo de simulación: 6:30 am - 1:30pm $\longrightarrow$ T = 7 horas = 25200 seg.
- Tiempo de llegada hasta ser atendido: Min=5seg, Max=30seg.
- Tiempo que tardan los clientes en ser atendidos:
    - Mujer: Min = 1 min= 60seg, Max = 5 min = 300 seg
    - Hombre: Min = 40 seg, Max = 2 min= 120 seg
- Consumo según el tipo de cliente:
    - Mujer: Min = 30 pesos, Max = 100 pesos
    - Hombre: Min = 20 pesos, Max = 80 pesos

Responder las siguientes preguntas basados en los datos del problema:
1. ¿Cuáles fueron los gastos de los hombres y las mujeres en 5 días de trabajo?.
2. ¿Cuál fue el consumo promedio de los hombres y mujeres?
3. ¿Cuál fue el número de personas atendidas por día?
4. ¿Cuál fue el tiempo de atención promedio?
5. ¿Cuánto fue la ganancia promedio de la cafetería en 5 días de trabajo y su respectiva rentabilidad?

## Respuesta a los ejercicios

### Ejercicio 1: 
Como ejemplo simple de una simulación de Monte Carlo, considere calcular la probabilidad de una suma particular del lanzamiento de tres dados (cada dado tiene valores del uno al seis). Además cada dado tiene las siguientes carácterísticas: el primer dado no está cargado (distribución uniforme todos son equiprobables); el segundo y tercer dado están cargados basados en una distribución binomial con parámetros (`n=5, p=0.5` y `n=5, p=0.2`). Calcule la probabilidad de que la suma resultante sea 7, 14 o 18. 

### Solución Lyha

In [8]:
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt


dado1 = np.random.randint(1,7,100000)
dado2 = st.binom(n=5,p=0.5, loc=1).rvs(size=100000)
dado3 = st.binom(n=5,p=0.2, loc=1).rvs(size=100000)

suma = dado1 + dado2 + dado3

In [9]:
p7 = len(suma[suma==7])/len(suma)
p14 = len(suma[suma==14])/len(suma)
p18 = len(suma[suma==18])/len(suma)
p7, p14, p18

(0.12674, 0.0136, 0.0)


- La probabilidad de que la suma resultante sea 7 es de aprox 12.67%
- La probabilidad de que la suma resultante sea 14 es de aprox 1.36%
- La probabilidad de que la suma resultante sea 18 es de aprox 0%


### Solución Daniel


 La respuesta es .........

## Ejercicio 2:
Cafetería Central

Premisas para la simulación:
- Negocio de alimentos que vende bebidas y alimentos.
- Negocio dentro del ITESO.
- Negocio en cafetería central.
- Tipo de clientes (hombres y mujeres).
- Rentabilidad del 60%.

### Objetivo
Realizar una simulación estimado el tiempo medio que se tardaran los clientes en ser atendidos entre el horario de 6:30 a 1 pm. Además saber el consumo. 
**Analizar supuestos y limitantes**
### Supuestos en simulación
Clasificación de clientes: 
- Mujer  = 1 $\longrightarrow$ aleatorio < 0.5
- Hombre = 0 $\longrightarrow$ aleatorio $\geq$ 0.5.

Condiciones iniciales:
- Todas las distrubuciones de probabilidad se supondrán uniformes.
- Tiempo de simulación: 6:30 am - 1:30pm $\longrightarrow$ T = 7 horas = 25200 seg.
- Tiempo de llegada hasta ser atendido: Min=5seg, Max=30seg.
- Tiempo que tardan los clientes en ser atendidos:
    - Mujer: Min = 1 min= 60seg, Max = 5 min = 300 seg
    - Hombre: Min = 40 seg, Max = 2 min= 120 seg
- Consumo según el tipo de cliente:
    - Mujer: Min = 30 pesos, Max = 100 pesos
    - Hombre: Min = 20 pesos, Max = 80 pesos

Responder las siguientes preguntas basados en los datos del problema:
1. ¿Cuáles fueron los gastos de los hombres y las mujeres en 5 días de trabajo?.
2. ¿Cuál fue el consumo promedio de los hombres y mujeres?
3. ¿Cuál fue el número de personas atendidas por día?
4. ¿Cuál fue el tiempo de atención promedio?
5. ¿Cuánto fue la ganancia promedio de la cafetería en 5 días de trabajo y su respectiva rentabilidad?

In [11]:
################## Datos del problema
d = 5
T =25200
T_at_min = 5; T_at_max = 30
T_mujer_min =60; T_mujer_max = 300
T_hombre_min = 40; T_hombre_max = 120
C_mujer_min = 30; C_mujer_max = 100
C_hombre_min = 20; C_hombre_max = 80

### Solución Lyha

In [12]:
def persona():
    x = np.random.randint(0,2)
    if x==1:
        return True
    else:
        return False 

In [13]:
def iteso():
    tiempo = 0 
    t=[]
    gasto=[]
    while tiempo < T:
        tiempoAt = np.random.randint(T_at_min ,T_at_max+1)
        tiempoM = np.random.randint(T_mujer_min, T_mujer_max+1)
        tiempoH = np.random.randint(T_hombre_min, T_hombre_max+1)

        if persona():    
                t.append(tiempoM+tiempoAt)
        else:
                t.append(tiempoH+tiempoAt)

        tiempo = sum(t)
    return t

In [14]:
sim_f1 = [iteso() for i in range(1000)]

In [23]:
clientes = [len(sim_f1[i]) for i in range(len(sim_f1))]
clientes_prom=sum(clientes)/len(clientes)

In [15]:
def iteso2():
    tiempo = 0 
    t=[]
    gasto=[]
    while tiempo < T:
        tiempoAt = np.random.randint(T_at_min ,T_at_max+1)
        tiempoM = np.random.randint(T_mujer_min, T_mujer_max+1)
        tiempoH = np.random.randint(T_hombre_min, T_hombre_max+1)

        if persona():    
                t.append(tiempoM+tiempoAt)
        else:
                t.append(tiempoH+tiempoAt)

        gastoM = np.random.randint(C_mujer_min, C_mujer_max+1)
        gastoH = np.random.randint(C_hombre_min, C_hombre_max+1)
        if persona():    
            gasto.append(gastoM)
        else:
            gasto.append(gastoH)
        tiempo = sum(t)
    return gasto

In [16]:
sim_f2 = [iteso2() for i in range(1000)]

1. ¿Cuáles fueron los gastos de los hombres y las mujeres en 5 días de trabajo?.

In [19]:
gasto_suma = [sum(sim_f2[i]) for i in range(len(sim_f2))]
gasto_prom = sum(gasto_suma)/len(gasto_suma)
gasto5_dias = gasto_prom*d
gasto5_dias

49234.045

Los gastos de los hombres y mujeres en 5 días de trabajo fueron de $49234.045

2. ¿Cuál fue el consumo promedio de los hombres y mujeres?


In [24]:
gasto_prom_final = gasto_prom/clientes_prom
gasto_prom_final

57.51103284739743

El consumo promedio es de $57.51

3. ¿Cuál fue el número de personas atendidas por día?


In [25]:
clientes_prom

171.216

Las personas atendidas por días son aprox. 171

4. ¿Cuál fue el tiempo de atención promedio?

In [26]:
tiempo=[sum(sim_f1[i]) for i in range(len(sim_f1))]
tiempo_prom = sum(tiempo)/len(tiempo)
tiempo_prom_final = tiempo_prom/clientes_prom
tiempo_prom_final

147.71510256050834

El tiempo promedio de atención es de 147.71 segundos 

5. ¿Cuánto fue la ganancia promedio de la cafetería en 5 días de trabajo y su respectiva rentabilidad?

In [27]:
rentabilidad= gasto5_dias*.6
gasto5_dias, rentabilidad

(49234.045, 29540.426999999996)

El gasto promedio en 5 días fue de $49234.045$ y la rentabilidad es de $29540.42$

### Solución Daniel


 La respuesta es .........